src/primitives/line/mod.rs

```diff
@@ -209,6 +209,21 @@ impl Line {
             return Intersection::Colinear;
         }
 
+        let outer_side = if denominator > 0 {
+            LineSide::Right
+        } else {
+            LineSide::Left
+        };
+
+        // Special case: If the two lines are almost parallel, return the average point between
+        // them.
+        if denominator.pow(2) < self.delta().dot_product(other.delta()) {
```

Comment:
You did mention that you don't know the math behind this, but can you explain how you came up with this formula?

Reply:
I had a test line from #471 set up in the simulator and printed a bunch of values whilst dragging the points around. I tried this formula and the spurs went away when I introduced it. It just seemed to work...